plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/catalog/hms/TrinoHiveCatalog.java

```diff
@@ -626,4 +632,34 @@ public Optional<CatalogSchemaTableName> redirectTable(ConnectorSession session,
         }
         return Optional.empty();
     }
+
+    private void validateSchemaTable(SchemaTableName schemaTableName)
+    {
+        if (metastore.getDatabase(schemaTableName.getSchemaName()).isEmpty()) {
+            throw new TrinoException(ICEBERG_INVALID_METADATA, format("Schema '%s' does not exist", schemaTableName.getSchemaName()));
```

Comment:
This is a generic user error,  not iceberg invalid metadata

Reply:
Added SCHEMA_NOT_FOUND